pkg/client/dbclient.go

```diff
@@ -1,51 +1,75 @@
 /*
-Copyright 2018 The KubeSphere Authors.
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-    http://www.apache.org/licenses/LICENSE-2.0
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
+
+ Copyright 2019 The KubeSphere Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
 */
 
 package client
 
 import (
 	"fmt"
+	"os"
+	"sync"
 
 	_ "github.com/go-sql-driver/mysql"
-	"github.com/golang/glog"
 	"github.com/jinzhu/gorm"
 )
 
-var dbClient *gorm.DB
-
-func NewDBClient() *gorm.DB {
-	conn := fmt.Sprintf("%s:%s@tcp(%s)/%s?charset=utf8mb4&parseTime=True&loc=Local", "", "", "", "")
+const (
+	dbHostEnv     = "DB_HOST"
+	dbUserEnv     = "DB_USER"
+	dbPasswordEnv = "DB_PASSWORD"
+	databaseEnv   = "DB_DATABASE"
+)
 
-	db, err := gorm.Open("mysql", conn)
+var (
+	dbHost       = "localhost:3306"
```

Comment:
I think command line arguments is better than environment variables in our case.
1. ps-grep friendly. we can use ps -ef | grep ks-apiserver to check all the arguments we passed
2. easy to debug. Developers can run binary alone on their local enviroment, command line option is better is this case.
3. avoid collisions with other program.